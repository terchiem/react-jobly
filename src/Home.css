.Home {
  text-align: center;
  margin: 80px auto;
}

.Home p {
  font-size: 3em;
}

.Home-login {
  display: block;
  width: 80px;
  padding: 10px;
  margin: 10px auto;
  background-color: #2469ff;
  color: #fff;
}