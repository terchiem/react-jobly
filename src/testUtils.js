import React from 'react';
import UserContext from './UserContext';

const testUser = {
  username: "test",
  first_name: "test",
  last_name: "test",
  email: "test@email.com",
  photo_url: null
}

const testToken = "test";

function UserProvider({ children, currentUser=testUser, token=testToken }) {
  return (
    <UserContext.Provider value={{currentUser, token}}>
      {children}
    </UserContext.Provider>
  )
};

export { UserProvider };