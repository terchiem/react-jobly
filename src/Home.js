import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import UserContext from './UserContext';
import './Home.css';

/** Displays a simple greeting to the user. Show a Login link if logged out. */

function Home() {
  const { currentUser } = useContext(UserContext); // TEMP: switch to user when implemented

  return (
    <div className="Home">
      <p>
        { 
          currentUser ? 
          `Welcome back, ${currentUser.first_name || currentUser.username}!` : 
          'Welcome to Jobly!'
        }
      </p>

      { currentUser ? null : <Link to="/login" className="Home-login">Login</Link>}
    </div>
  );
}

export default Home;