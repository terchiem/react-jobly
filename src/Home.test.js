import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Home from './Home';
import { UserProvider } from './testUtils';

test('renders without crashing', () => {
  render(
    <MemoryRouter>
      <UserProvider>
        <Home />
      </UserProvider>
    </MemoryRouter>
  );
});

test('matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Home />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('matches snapshot when logged out', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <Home />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});