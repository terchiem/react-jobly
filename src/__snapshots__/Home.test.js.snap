// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      Jobly
    </h1>
    <p>
      Welcome back, test!
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      Jobly
    </h1>
    <p>
      <a
        class="Home-login"
        href="/login"
      >
        Login
      </a>
    </p>
  </div>
</DocumentFragment>
`;
